controller/backup_controller.go

```diff
@@ -452,7 +452,7 @@ func (bc *BackupController) isResponsibleFor(b *longhorn.Backup, defaultEngineIm
 
 	isResponsible := isControllerResponsibleFor(bc.controllerID, bc.ds, b.Name, "", b.Status.OwnerID)
 
-	readyNodesWithReadyEI, err := bc.ds.ListReadyNodesWithReadyEngineImage(defaultEngineImage)
+	readyNodesWithReadyEI, err := bc.ds.ListReadyNodesWithEngineImage(defaultEngineImage)
```

Comment:
We can remove `ListReadyNodesWithReadyEngineImage` if no ones use it.

Also, the below code seems unnecessary. I don't see anything it is trying to achieve. 

```go
	// No node in the system has the default engine image in ready state
	if len(readyNodesWithReadyEI) == 0 {
		return false, nil
	}
```

Reply:
Yes, you are right
I am just thinking why do we add this the first time
it is added in this PR: https://github.com/longhorn/longhorn-manager/pull/1038/files
but after review that PR, this seems unnecessary
cause the following code will check if the current node has ready image `currentNodeEngineAvailable`
and if `currentNodeEngineAvailable = False`, the function will always return False 
There is no need to check if "all nodes" have ready image or not
```
	currentOwnerEngineAvailable, err := bc.ds.CheckEngineImageReadiness(defaultEngineImage, b.Status.OwnerID)
	if err != nil {
		return false, err
	}
	currentNodeEngineAvailable, err := bc.ds.CheckEngineImageReadiness(defaultEngineImage, bc.controllerID)
	if err != nil {
		return false, err
	}

	isPreferredOwner := currentNodeEngineAvailable && isResponsible
	continueToBeOwner := currentNodeEngineAvailable && bc.controllerID == b.Status.OwnerID
	requiresNewOwner := currentNodeEngineAvailable && !currentOwnerEngineAvailable
        return isPreferredOwner || continueToBeOwner || requiresNewOwner, nil
```

cc @shuo-wu

Reply:
Fixed, let me retest the previous issue https://github.com/longhorn/longhorn/issues/2960 (Fix backup target status.available always false) as well

Reply:
both cases have passed locally
- cannot create backup when engine is not fully deployed
- target won't be reconcile